token/token.go

```diff
@@ -613,6 +617,44 @@ func IsNeedQuoted(value string) bool {
 	return false
 }
 
+// shouldQuoteRune returns true if the rune should be quoted.
+// excludes all runes in the Basic Multilingual Plane and all Emoticons
+func shouldQuoteRune(r rune) bool {
+	// Basic Multilingual Plane, Letters and Emoji
+	if r < 0x1000 || unicode.IsLetter(r) || isEmoticon(r) {
+		return false
+	}
+	return true
+}
+
+func isEmoticon(r rune) bool {
```

Comment:
Is there any information I can reference to determine if this process is correct?

Reply:
@goccy good question, I found this function in other code samples on GitHub and StackOverflow and have been using it successfully in oh-my-posh for the past year to correctly avoid escaping emoticons. It's basically [this](https://unicode.org/emoji/charts/full-emoji-list.html) list in a different format.

Reply:
Thank you . Please mention this link in the source code comments as well .